java/src/server/plugin/PluginLoader.java

```diff
@@ -1,35 +1,55 @@
 package server.plugin;
+
+import com.esotericsoftware.yamlbeans.YamlReader;
+
 import java.io.File;
-import java.net.MalformedURLException;
+import java.io.FileReader;
 import java.net.URL;
 import java.net.URLClassLoader;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Objects;
+import java.util.*;
 import java.util.stream.Collectors;
 
-import static java.util.Arrays.stream;
+import static server.plugin.PluginConfig.PluginType.PERSISTENCE;
 
 public class PluginLoader implements IPluginLoader {
 
     /**
      * @inheritDoc
      */
     @Override
-    public List<IPlugin> loadConfig(List<PluginConfig> configs, File folder) {
-        List<String> neededPlugins = configs.stream()
-                .map(o -> o.getName().toLowerCase())
+    public List<IPlugin> loadConfig(List<PluginConfig> pluginConfigs, File folder) {
+        // neededPlugins
+        List<String> neededPlugins = pluginConfigs.stream()
```

Comment:
I figured you'd be interested. The parameters for `Collectors.toMap` are two functions. The first and second accept an item of the stream, then output the key and value for that item respectively.